src/devices/Charlieplex/CharlieplexSegment.cs

```diff
@@ -137,15 +141,16 @@ public void Write(int node, PinValue value, int duration = 0)
 
             if (duration > 0)
             {
-                DisplaySegment(TimeSpan.FromMilliseconds(duration));
+                using CancellationTokenSource cts = new(duration);
```

Comment:
NIT: I know you are not adding this, but duration should be a timespan, not an int

Reply:
Will fix that.